src/librustdoc/html/render/mod.rs

```diff
@@ -2346,9 +2355,57 @@ fn sidebar_trait(cx: &Context<'_>, buf: &mut Buffer, it: &clean::Item, t: &clean
     buf.push_str("</section>")
 }
 
+/// Returns the filtered implementation lists for the primitive reference type. It works as follows:
```

Comment:
nit: no need to say "it works as follows"

How about:

"Returns the list of implementations for the primitive reference type, filtering out any implementations that are on concrete or partially generic types, only keeping implementations of the form `impl<T> Trait for &T`

Reply:
:+1: